File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/data/DataFloat32Dims.java
package tasks.data;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Float32;

@JavaTaskInfo
public class DataFloat32Dims extends JavaTask {

	@InputPortInfo(name = "In", dataType = Float32.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Float32.class)
	public OutputPort out;

	@Override
	public void execute(int grp) {
		Float32 inFloat32 = in.getData(Float32.class);
		float[] infloat32 = inFloat32.getValues();
		float[] outfloat = new float[inFloat32.getNumberOfElements()];
		for(int i = 0; i < inFloat32.getNumberOfElements(); i++)
			outfloat[i] = infloat32[i]*2;
			
		out.putData(new Float32(inFloat32.getDimensions(), outfloat));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/data/DataFloat32Dims.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Float32 import Float32
from operator import add

@PythonTaskInfo
class DataFloat32Dims(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Float32)
        self.in2 = InputPort(name="in2", data_type=Float32)
        self.out = OutputPort(name="out", data_type=Float32)

    def execute(self, grp):
        in1Data = self.in1.get_data(Float32)
        in2Data = self.in2.get_data(Float32)
        ret = list(map(add, in1Data.values, in2Data.values))
        self.out.put_data(Float32(ret, in1Data.dims))
